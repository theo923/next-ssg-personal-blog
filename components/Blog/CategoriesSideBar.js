import React from 'react';
import Link from 'next/link';

const CategoriesSideBar = ({categories}) => {

    const CategoriesGenerator = (tag) => {
        return (
            <div key={tag} className="flex flex-wrap 2xl:flex-col px-6 ">
                <Link href={`/category/${tag}`} >
                    <div className="flex items-center my-2 mt-6  bg-white cursor-pointer hover:bg-gray-200 rounded-lg text-gray-700 hover:text-blue-600">
                        <svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 24 24" fill="none" stroke="#000000" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"><path d="M9 18l6-6-6-6"/></svg>
                        <div className="font-hindSiliguri px-2 text-lg">{tag}</div>
                    </div>
                </Link>
            </div>
        )
    }

    return (
        <div className='block sticky top-14 overflow-hidden' style={{maxHeight: 'calc(100vh - 148px)', flex: '0 100000 250px'}}>
            <div className='mt-12 mx-6 mb-6 2xl:ml-3 mr-4 md:mr-6 xl:mr-8 shadow-xl'>
                <div className="py-4 2xl:w-64 bg-white rounded-lg shadow-lg">
                    <h1 className="font-prozaLibre mx-10 lg:mt-10 text-2xl lg:text-3xl font-semibold text-gray-800">Categories</h1>
                    {!categories ? null : categories.map(tag => CategoriesGenerator(tag))}
                </div>
            </div>
        </div>
    )
    
}

export default CategoriesSideBar;
