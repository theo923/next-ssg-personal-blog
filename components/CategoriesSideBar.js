import React from 'react';
import Link from 'next/link';

const CategoriesSideBar = ({categories}) => {

    const CategoriesGenerator = (tag) => {
        return (
            <div>
                <Link href={`/category/${tag}`} >
                    <div key={tag} className="flex items-center py-2 mt-6  bg-white cursor-pointer hover:bg-gray-200 rounded-lg text-gray-700 hover:text-blue-600 dark:text-gray-200">
                        <svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 24 24" fill="none" stroke="#000000" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"><path d="M9 18l6-6-6-6"/></svg>
                        <div className="font-hindSiliguri px-2 text-lg">{tag}</div>
                    </div>
                </Link>
            </div>
        )
    }

    return (
        <div className='block sticky top-14 overflow-x-auto overflow-y-auto' style={{maxHeight: 'calc(100vh - 148px)', flex: '0 100000 250px'}}>
            <div className='mt-12 mx-6 mb-6 2xl:ml-3 mr-4 md:mr-6 xl:mr-8 shadow-xl'>
                <div className="pb-4 w-64 mx-auto overflow-hidden bg-white rounded-lg shadow-lg dark:bg-gray-800">
                    <h1 className="font-prozaLibre px-10 mt-10 text-3xl font-semibold text-gray-800 dark:text-white">Categories</h1>
                    <div className="px-6 py-4 ">
                        {!categories ? null : categories.map(tag => CategoriesGenerator(tag))}
                    </div>
                </div>
            </div>
        </div>
    )
    
}

export default CategoriesSideBar;
