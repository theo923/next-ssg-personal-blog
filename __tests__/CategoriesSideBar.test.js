import { shallow } from 'enzyme';
import React from 'react';
import CategoriesSideBar from '../components/Blog/CategoriesSideBar';

describe('CategoriesSideBar test', () => {

    it("expect CategoriesSideBar component is rendered without crashing", () => {
        const wrapper = shallow(<CategoriesSideBar />);   
    });

    it("expect CategoriesSideBar component is rendered && match with the Snapshot", () => {
        const wrapper = shallow(<CategoriesSideBar />);   
        expect(shallow(<CategoriesSideBar />)).toMatchSnapshot();
    });
    
});
