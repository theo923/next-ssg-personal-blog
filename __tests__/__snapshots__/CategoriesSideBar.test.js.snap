// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesSideBar test expect CategoriesSideBar component is rendered && match with the Snapshot 1`] = `
<div
  className="block sticky top-14 overflow-hidden"
  style={
    Object {
      "flex": "0 100000 250px",
      "maxHeight": "calc(100vh - 148px)",
    }
  }
>
  <div
    className="mt-12 mx-6 mb-6 2xl:ml-3 mr-4 md:mr-6 xl:mr-8 shadow-xl"
  >
    <div
      className="py-4 2xl:w-64 bg-white rounded-lg shadow-lg dark:bg-gray-800"
    >
      <h1
        className="font-prozaLibre mx-10 lg:mt-10 text-2xl lg:text-3xl font-semibold text-gray-800 dark:text-white"
      >
        Categories
      </h1>
      <div
        className="flex flex-wrap 2xl:flex-col px-6 lg:py-4 "
      />
    </div>
  </div>
</div>
`;
